core/src/test/java/org/hibernate/ogm/backendtck/id/TableIdGeneratorInheritanceTest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.backendtck.id;
+
+import java.util.UUID;
+import javax.persistence.Entity;
+import javax.persistence.EntityManager;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.Inheritance;
+import javax.persistence.Table;
+import org.hibernate.annotations.GenericGenerator;
+import org.hibernate.ogm.utils.jpa.OgmJpaTestCase;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static javax.persistence.InheritanceType.TABLE_PER_CLASS;
+import static org.fest.assertions.Assertions.assertThat;
+
+/**
+ * Test that @Inheritance(strategy = TABLE_PER_CLASS) with generated id work correct
+ */
+public class TableIdGeneratorInheritanceTest extends OgmJpaTestCase {
+
+	private EntityManager em;
+
+	@Before
+	public void setUp() {
+		em = getFactory().createEntityManager();
+	}
+
+	@After
+	public void tearDown() {
+		em.close();
+	}
+
+	@Test
+	public void testTableIdGenerator() {
+		em.getTransaction().begin();
+		Truck truck = new Truck();
+		em.persist( truck );
+		em.getTransaction().commit();
+
+		em.clear();
+
+		em.getTransaction().begin();
+
+		Truck loadedTruck = em.find( Truck.class, truck.getId() );
+
+		assertThat( loadedTruck ).isNotNull().as( "truck not loaded" );
+		assertThat( loadedTruck.getId() ).isNotNull().as( "truck id not loaded" );
+		assertThat( loadedTruck.getId() ).isEqualTo( truck.getId() )
+				.as( "loaded truck id is not equals saved truck id" );
```

Comment:
Let's simplify the assertions:
```
		assertThat( loadedTruck ).isNotNull();
		assertThat( loadedTruck.getId() ).isEqualTo( truck.getId() );
```

No need to check that getId is not null, it's already done when you do the isEqualTo. The  messages aren't very helpful in this case and only make the code base bigger.

Reply:
If I understand isEqualTo right, then it would be true if all id will be null.
But it's not right, that's why I've added this assert.
If you stil think it's excess, then I remove it?

Reply:
If both Id are null then it works because they are, in fact, equals :)

But you raise a valid point about checking that the id has been generated, I would prefer to have a separate test method for this and you should probably check the truck.getId() is not null after session.persist(). Having it in a separate test method make it easier for other programmers to understand what's going on.

Reply:
Ok, I'll devide test to two cases.